src/Adapter/Image/ImageRetriever.php

```diff
@@ -246,14 +246,42 @@ public function getImage($object, $id_image)
         $large = end($urls);
         $medium = $urls[$keys[ceil((count($keys) - 1) / 2)]];
 
-        return [
+        $result = [
             'bySize' => $urls,
             'small' => $small,
             'medium' => $medium,
             'large' => $large,
             'legend' => !empty($object->meta_title) ? $object->meta_title : $object->name,
             'id_image' => $id_image,
         ];
+
+        if ($this->isMultipleImageFormatFeatureActive) {
+            $result['original'] = $this->getOriginalImage($getImageURL, $id_image, $rewrite);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param string $getImageURL
+     * @param int $imageId
+     * @param string $rewrite
+     *
+     * @return array
+     */
+    private function getOriginalImage(string $getImageURL, int $imageId, string $rewrite): array
+    {
+        $originalImagePath = $this->link->$getImageURL($rewrite, $imageId);
+        $originalDimensions = getimagesize($originalImagePath);
```

Comment:
I am really not sure about this @Hlavtox - I think it can drastically decrease performance, especially if you use some CDN. We'd need a better way of getting image size:
https://stackoverflow.com/a/4635991/2895156

but it is still quite slow. Problems with performance using this function happened in other projects as well:
https://www.drupal.org/project/caption_filter/issues/2299925

Reply:
There must have been some communication mistake, we need to check the local path.

```
$original_path = implode(DIRECTORY_SEPARATOR, [
            $imageFolderPath,
            $id_image . '.jpg',
        ]);

        $original_dimensions = getimagesize($original_path);
        $original = [
            'url' => $this->link->$getImageURL(
            isset($object->link_rewrite) ? $object->link_rewrite : $object->name,
            $id_image
        ),
            'width' => $original_dimensions[0],
            'height' => $original_dimensions[1],
        ];
​
        if ($original_dimensions[0] < 1 || $original_dimensions[1] < 1) {
            $original = [
                'url' => $large['url'],
                'width' => $large['width'],
                'height' => $large['height'],
            ];
        }
```

Reply:
I tested the performance and it takes 0.70418691635132 ms to getimagesize of 64000 photos.

Reply:
probably needs some checks to avoid notices in case the file is missing or corrupted

Reply:
That is handled by the getimagesize, it returns null for the size if I remember correctly

Reply:
Nope. It [generates errors](https://www.php.net/manual/en/function.getimagesize#refsect1-function.getimagesize-errors) when file is invalid